include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -83,6 +83,98 @@ namespace internal
 
 
 
+    /**
+     * Generate evalution points on a simplex with arbitrary number of
+     * subdivisions.
+     */
+    template <int dim>
+    inline std::vector<Point<dim>>
+    generate_simplex_evaluation_points(const unsigned int n_subdivisions)
+    {
+      (void)n_subdivisions;
+
+      Assert(false, ExcNotImplemented());
+
+      return {};
+    }
+
+
+
+    /**
+     * Helper function to create evaluation points recursivly.
+     */
+    inline void
+    generate_simplex_evaluation_points_recursivly(
+      const std::vector<Point<2>> &bounding_vertices,
+      const unsigned int           n_subdivisions,
+      std::vector<Point<2>> &      evalution_points)
+    {
+      if (n_subdivisions == 0)
+        {
+          evalution_points.push_back(bounding_vertices[0]);
+          return;
+        }
+
+      for (const auto &p : bounding_vertices)
+        evalution_points.push_back(p);
+
+      if (n_subdivisions == 1)
+        return;
+
+      const auto process = [&](const Point<2> &p0, const Point<2> &p1) {
```

Comment:
Maybe call this function `generate_inbetween_points()`, which I think would make it more obvious what it does.

Reply:
I think it would also be good to write a comment that explains what the function does. It has a bit of a complicated logic: It *both* returns an array, but *also* appends what it returns to the existing `evaluation_point`.